src/Disks/DiskObjectStorage/ObjectStorages/AzureBlobStorage/AzureObjectStorage.cpp

```diff
@@ -189,7 +189,6 @@ void AzureObjectStorage::listObjects(const std::string & path, RelativePathsWith
         if (client_ptr->IsClientForDisk())
             ProfileEvents::increment(ProfileEvents::DiskAzureListObjects);
 
-        blob_list_response = client_ptr->ListBlobs(options);
```

Comment:
To be honest I do not understand how it worked before at all, it looks like because of this `blob_list_response` override inside the loop we always listed only the first page and never went to the next one?